src/dmd/aggregate.d

```diff
@@ -134,7 +134,9 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
     Scope* newScope(Scope* sc)
     {
         auto sc2 = sc.push(this);
-        sc2.stc &= STC.safe | STC.trusted | STC.system;
+        sc2.stc &= (
+                STC.safe | STC.trusted | STC.system |
+                STC.nogc | STC.pure_ | STC.nothrow_);
```

Comment:
This surely can't be the right fix. It just seems that the scope for aggregates is implicitly created with all the attributes.

Reply:
The current system is:
* Make a new child scope, which inherits everything from the parent scope
* Prune the stuff that doesn't make sense

That means that this is in fact the right fix.